geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionAdvisorDUnitTest.java

```diff
@@ -120,4 +98,45 @@ public void testGenericAdvice() {
     }
     assertEquals(expected, advisor.adviseGeneric());
   }
+
+  @Test
+  public void advisorIssuesSevereAlertForStateFlush() throws Exception {
+    final long membershipVersion = advisor.startOperation();
+    advisor.forceNewMembershipVersion();
+
+    final Logger logger = mock(Logger.class);
+    final Exception exceptionHolder[] = new Exception[1];
+    Thread thread = new Thread(() -> {
+      try {
+        advisor.waitForCurrentOperations(logger, 2000, 4000);
+      } catch (RuntimeException e) {
+        synchronized (exceptionHolder) {
+          exceptionHolder[0] = e;
+        }
+      }
+    });
+    thread.setDaemon(true);
+    thread.start();
+
+    try {
+      Awaitility.await().atMost(15, TimeUnit.SECONDS).until(() -> {
+        verify(logger, atLeastOnce()).warn(isA(String.class), isA(Long.class));
+      });
+      Awaitility.await().atMost(15, TimeUnit.SECONDS).until(() -> {
+        verify(logger, atLeastOnce()).fatal(isA(String.class), isA(Long.class));
+      });
```

Comment:
Would it be possible for the test, after the fatal message is confirmed, to then make a change so the opCount goes to zero and you can then confirm the new info message is logged? You could also then assert that the thread should terminate naturally instead of needing to be interrupted. Seems like if you see this info message and the thread is still running after joining for 10000 then something is wrong

Reply:
Good ideas for the test.  I'll implement them all.